src/group_theory/free_group.lean

```diff
@@ -864,3 +867,87 @@ fintype.subtype (list.to_finset $ red.enum L₁) $
 end reduce
 
 end free_group
+
+
+/-- `is_free_group G` means that `G` isomorphic to a free group. -/
+class is_free_group (G : Type u) [group G] :=
+  ( generators : Type u )
+  ( mul_equiv : free_group generators ≃* G )
+
+instance (X : Type*) : is_free_group (free_group X) :=
+  { generators := X, mul_equiv := mul_equiv.refl _ }
+
+namespace is_free_group
+
+variables (G : Type*) [group G] [is_free_group G]
+
+/-- Any free group is isomorphic to "the" free group. -/
+@[simps] def to_free_group : G ≃* free_group (generators G) := is_free_group.mul_equiv.symm
+
+variable {G}
+
+/-- The canoncial injection of G's generators into G -/
+def of : generators G → G := is_free_group.mul_equiv.to_fun ∘ free_group.of
+
+@[simp] lemma of_eq_free_group_of {A : Type u} : (@of (free_group A) _ _ ) = free_group.of := rfl
+
+variables {H : Type*} [group H]
+
+/-- The equivalence between functions on the generators and group homomorphisms from a free group
+given by those generators. -/
+@[simps symm_apply]
+def lift : (generators G → H) ≃ (G →* H) :=
+  free_group.lift.trans
+  { to_fun := λ f, f.comp (@is_free_group.mul_equiv G _ _).symm.to_monoid_hom,
+    inv_fun := λ f, f.comp (@is_free_group.mul_equiv G _ _).to_monoid_hom,
+    left_inv := λ f, by { ext, simp, },
+    right_inv := λ f, by { ext, simp, }, }
```

Comment:
```suggestion
def lift : (generators G → H) ≃ (G →* H) :=
free_group.lift.trans
  { to_fun := λ f, f.comp (@is_free_group.mul_equiv G _ _).symm.to_monoid_hom,
    inv_fun := λ f, f.comp (@is_free_group.mul_equiv G _ _).to_monoid_hom,
    left_inv := λ f, by { ext, simp, },
    right_inv := λ f, by { ext, simp, }, }
```

Reply:
(Although probably reducing the indenting of the entire `{ ... }` block is more common mathlib style.)